src/duckdb/src/common/types/value.cpp

```diff
@@ -1056,17 +1082,21 @@ T Value::GetValueInternal() const {
 	case LogicalTypeId::TIME_TZ:
 		return Cast::Operation<dtime_tz_t, T>(value_.timetz);
 	case LogicalTypeId::TIMESTAMP:
-	case LogicalTypeId::TIMESTAMP_TZ:
 		return Cast::Operation<timestamp_t, T>(value_.timestamp);
+	case LogicalTypeId::TIMESTAMP_SEC:
+		return Cast::Operation<timestamp_sec_t, T>(value_.timestamp_s);
+	case LogicalTypeId::TIMESTAMP_MS:
+		return Cast::Operation<timestamp_ms_t, T>(value_.timestamp_ms);
+	case LogicalTypeId::TIMESTAMP_NS:
+		return Cast::Operation<timestamp_ns_t, T>(value_.timestamp_ns);
+	case LogicalTypeId::TIMESTAMP_TZ:
+		return Cast::Operation<timestamp_tz_t, T>(value_.timestamp_tz);
 	case LogicalTypeId::UTINYINT:
 		return Cast::Operation<uint8_t, T>(value_.utinyint);
 	case LogicalTypeId::USMALLINT:
 		return Cast::Operation<uint16_t, T>(value_.usmallint);
 	case LogicalTypeId::UINTEGER:
 		return Cast::Operation<uint32_t, T>(value_.uinteger);
-	case LogicalTypeId::TIMESTAMP_MS:
-	case LogicalTypeId::TIMESTAMP_NS:
-	case LogicalTypeId::TIMESTAMP_SEC:
 	case LogicalTypeId::UBIGINT:
 		return Cast::Operation<uint64_t, T>(value_.ubigint);
```

Comment:
@taniabogatsch: This particular chunk (upstream: https://github.com/duckdb/duckdb/pull/14818) fails one of our test cases -- a roundtrip DuckDB -> Arrow -> DuckDB followed by a query. Reverting only this chunk keeps the test happy.

Test:

https://github.com/duckdb/duckdb-r/blob/d3194b2af2bd14159083ee304b0f2c9f1a73e9c7/tests/testthat/test-register_arrow.R#L328-L360 

The error is:

```
Error: Not implemented Error: Unimplemented type for cast (INVALID -> INT64)
```

Test run: https://github.com/duckdb/duckdb-r/actions/runs/12485542302/job/34844461922#step:17:399

Raising this early in case this rings a bell. I'll work on a better reproducer.

Reply:
This was a sloppiness in the R interface code, mixing `timestamp_t` and `timestamp_tz_t` . Fixing the sloppiness fixes everything.